### ISSUE TITLE
Argv list not endswith NULL in stack initialization

### ISSUE THREAD

**Issue by mohanson** [nervosnetwork/ckb-vm#98]
I found that ckb-vm's argv has not endswith null, and then I did a simple test.

The C Standard 5.1.2.2.1/2 says:

> argv[argc] shall be a null pointer.

so create a c program:

```c
#include <stddef.h>

int main(int argc,  char *argv[]) {
	if (argv[argc] == NULL) {
		return 0;
	}
	return 1;
}
```

When built by gcc, and run on Linux, it returns "0".

When built by riscv-unknown-elf-gcc, run on ckb-vm, it returns "1".

You might add a NULL here: https://github.com/nervosnetwork/ckb-vm/blob/develop/src/machine/mod.rs#L160

**Comment by xxuejie** [comment]
We could make this change but has to do it in a non-breaking way, I will give it a thought on how we can achieve this, thanks for the report

**Comment by xxuejie** [comment]
Fixed in #99
